src/modules/jcms_notifications/src/FetchArticleService.php

```diff
@@ -63,6 +63,32 @@ public function getArticle(string $id) {
   }
 
   /**
+   * Gets the IDs for every article in Lax.
+   *
+   * @return array
+   */
+  public function getAllArticleIDs() {
+    $ids = [];
+    $endpoint = Settings::get('jcms_all_articles_endpoint');
+    if ($endpoint) {
+      $auth_key = 'jcms_article_auth_unpublished';
+      $response = $this->client->get($endpoint, ['auth' => Settings::get($auth_key), 'http_errors' => FALSE]);
+      if ($response instanceof ResponseInterface) {
+        $body = $response->getBody()->getContents();
+        $json = json_decode($body, TRUE);
+        if (isset($json['items'])) {
```

Comment:
If you need to introduce batch processing then it may just be as good to handle the results on a page in each batch.